frontend/dispatcher.lua

```diff
@@ -94,84 +299,48 @@ function Dispatcher:init()
     Dispatcher.initialized = true
 end
 
---[[--
-Add a submenu to edit which items are dispatched
-arguments are:
-    1) self
-    2) the table representing the submenu (can be empty)
-    3) the name of the parent of the settings table (must be a child of self)
-    4) the name of the settings table
-example usage:
-    Dispatcher.addSubMenu(self, sub_items, "profiles", "profile1")
---]]--
-function Dispatcher:addSubMenu(menu, location, settings)
-    if not Dispatcher.initialized then Dispatcher:init() end
-    table.insert(menu, {
-        text = _("None"),
-        separator = true,
-        checked_func = function()
-            return next(self[location][settings]) == nil
-        end,
-        callback = function(touchmenu_instance)
-            self[location][settings] = {}
-            if touchmenu_instance then touchmenu_instance:updateItems() end
-        end,
-    })
-    for k, v in pairs(settingsList) do
-            if settingsList[k].category == "none" then
-                table.insert(menu, {
-                   text = settingsList[k].title,
-                   checked_func = function()
-                   return self[location][settings] ~= nil and self[location][settings][k] ~= nil
-                   end,
-                   callback = function(touchmenu_instance)
-                      if self[location][settings] ~= nil
-                      and self[location][settings][k] then
-                          self[location][settings][k] = nil
-                      else
-                          self[location][settings][k] = true
-                      end
-                      if touchmenu_instance then touchmenu_instance:updateItems() end
-                  end,
-               })
-            elseif settingsList[k].category == "toggle" then
+function Dispatcher.addItem(caller, menu, location, settings, section)
+    for _,k in ipairs(dispatcher_menu_order) do
+        if settingsList[k][section] == true and
+        (settingsList[k].condition == nil or settingsList[k].condition) then
```

Comment:
Indentation (even 2 x 4 to make it stand out of the next body). On on a single line, it's not too long.